packages/near-api-js/src/account_multisig.ts

```diff
@@ -381,6 +396,61 @@ export class Account2FA extends AccountMultisig {
         });
     }
 
+    /**
+     * Returns `false` if multisig cannot be disabled by calling disable() directly
+     * i.e. account has too many LAKs to delete and re-create as FAKs for a single transaction
+     */
+    async canDisableMultisig() {
+        const keys = await this.get2faLimitedAccessKeys();
+
+        // there are max 4 actions required to disable 2FA on top of the key conversion (2 actions per key)
+        // (100 - 4) / 2 = 48 = LAK_CONVERSION_LIMIT - 2
+        return keys.length <= (LAK_CONVERSION_LIMIT - 2);
+    }
+
+    /**
+     * Converts multisig LAKs (excluding the active key in the wallet, provided here as the
+     * `signingPublicKey` argument) back to FAKs when there are too many key conversions to be
+     * executed within the same transaction.
+     *
+     * Note that this undermines the security provided by 2FA by converting existing multisig LAKs into FAKs
+     * capable of bypassing the 2FA prompt when signing transactions outside the wallet (transactions signed
+     * in the wallet will continue to require 2FA approval). This method should only be used to convert LAKs
+     * as a way to make 2FA disabling possible.
+     *
+     * To be deprecated after 2FA migration; this addresses a specific issue with multisig deployed via wallet.
+     * @param signingPublicKey the public key used to sign transactions in the wallet
+     */
+    async batchConvertKeys(signingPublicKey: string) {
+        if (!signingPublicKey || !PublicKey.fromString(signingPublicKey)) {
+            throw new Error('invalid public key');
+        }
+
+        await this.validateMultisigState();
+
+        let batch = 0;
+        const keysToConvert = (await this.get2faLimitedAccessKeys())
+            .filter(({ public_key }) => public_key !== signingPublicKey);
+
+        while (batch < keysToConvert.length) {
```

Comment:
Why not do this only while `canDisableMultisig` is `false`? There would be less prompts that way and UX would be better

Reply:
Good point. After reviewing the workflow in the wallet I think it would be better to just call the `disable` method in this method directly once the number of keys is under the threshold.